moonraker/components/power.py

```diff
@@ -267,6 +263,14 @@ def __init__(self, config: ConfigHelper) -> None:
         self.locked_while_printing = config.getboolean(
             'locked_while_printing', False)
         self.off_when_shutdown = config.getboolean('off_when_shutdown', False)
+        self.off_when_shutdown_delay = 0.
+        if self.off_when_shutdown:
+            self.off_when_shutdown_delay = config.getfloat(
+                'off_when_shutdown_delay', 0.)
+            if self.off_when_shutdown_delay < 0.:
```

Comment:
The config "getfloat" and "getint" methods now accept optional `minval`, `maxval`, `above` and `below` arguments so it is no longer necessary to implement range checks.  For example:
```python
self.off_when_shutdown_delay = config.getfloat('off_when_shutdown_delay', 0., minval=0.)
```
Should accomplish this.  There are likely several places in the code that I need to switch over to this method, however I think its best to start using this going forward.